File: app/src/main/java/com/divercity/app/features/profile/selectcompany/company/CompanyPaginatedRepository.java
package com.divercity.app.features.profile.selectcompany.company;

import android.support.annotation.Nullable;

import com.divercity.app.core.utils.Listing;
import com.divercity.app.data.entity.company.CompanyResponse;

/**
 * Created by lucas on 01/10/2018.
 */

public interface CompanyPaginatedRepository {

    Listing<CompanyResponse> fetchCompanies(@Nullable String query);

    void retry();

    void refresh();

    void clear();

}

File: app/src/main/java/com/divercity/android/features/company/selectcompany/base/CompanyPaginatedRepository.kt
package com.divercity.android.features.company.selectcompany.base

import com.divercity.android.core.base.datasource.BaseDataSourceRepository
import com.divercity.android.core.base.usecase.Params
import com.divercity.android.core.base.usecase.UseCase
import com.divercity.android.data.entity.company.response.CompanyResponse
import com.divercity.android.features.company.selectcompany.base.usecase.FetchCompaniesUseCase
import javax.inject.Inject

/**
 * Created by lucas on 01/10/2018.
 */

class CompanyPaginatedRepository @Inject
internal constructor(private val fetchCompaniesUseCase: FetchCompaniesUseCase) :
    BaseDataSourceRepository<CompanyResponse>() {

    override fun getUseCase(): UseCase<List<CompanyResponse>, Params> {
        return fetchCompaniesUseCase
    }
}
